### PR TITLE
Add test case with conditional compilation

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to https://github.com/SonarSource/sonar-dotnet/issues/4540

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNestedBlock.cs [hunk 1]
```diff
@@ -81,6 +82,14 @@ void F1(bool b)
         void F2()
         {
         }
+
+        // See: https://github.com/SonarSource/sonar-dotnet/issues/4540
+        public void ConditionalCompilation(object obj)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNestedBlock.cs:87]: If I understood correctly this rule looks only at nested blocks. So, the FP can be reproduced with the following code example:
```
        // See: https://github.com/SonarSource/sonar-dotnet/issues/4540
        public void ConditionalCompilation(object obj)
        {
            if (obj != null)
            {
#if DEBUG
                Trace.WriteLine("message");
#endif
            }
        }
```

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Please try the suggested reproducer.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNestedBlock.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNestedBlock.cs:87]: Thanks. I've took the example from the issue without reading the rule description thoroughly.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM! I think we should tell the reporter of the issue that we managed to create a reproducer.

**Comment by costin-zaharia-sonarsource** [comment]
> I think we should tell the reporter of the issue that we managed to create a reproducer.

Yes, I've done that already.

**Pull request merged**